### PR TITLE
[TASK] Have `comsumeWhiteSpace()` return the consumed

### PR DESCRIPTION
**Pull request by JakeQZ**
Instead, have the comments array passed by reference to be appended to.  This avoids various methods having to do an `array_merge`.  Also, knowing if any whitespace was consumed will be needed for #1471.

### CODE REVIEW

File: src/Parsing/ParserState.php [hunk 1]
```diff
@@ -197,12 +197,12 @@ public function parseCharacter(bool $isForIdentifier): ?string
      * This method may change the state of the object by advancing the internal position;
      * it does not simply 'get' a value.
```
Comment (JakeQZ) [inline src/Parsing/ParserState.php:202]: This is still required, despite the change.  I'm slightly disappointed that Stan can't work this out, as he is so good at everything else.

**Comment by JakeQZ** [comment]
I hope [David Gilmour](https://en.wikipedia.org/wiki/David_Gilmour) will appreciate this.

File: src/Parsing/ParserState.php (hunk 1 above)
Comment (oliverklee) [reply src/Parsing/ParserState.php:202]: It might at higher levels (with PHP-CSS-Parser, we're still only at level 5).

**Comment by oliverklee** [comment]
> I hope [David Gilmour](https://en.wikipedia.org/wiki/David_Gilmour) will appreciate this.

I don't understand the connection to this PR. :cry: Can you help me out?

File: src/Parsing/ParserState.php [hunk 2]
```diff
@@ -197,12 +197,12 @@ public function parseCharacter(bool $isForIdentifier): ?string
...
      */
```
Comment (oliverklee) [inline src/Parsing/ParserState.php:203]: Maybe we can add a sentence or two to the method description explaining what the method does and what the return value means.

File: tests/Unit/Parsing/ParserStateTest.php [hunk 3]
```diff
@@ -157,4 +157,105 @@ public function consumeIfComesUpdatesLineNumber(): void
...
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider provideWhitespace
+     */
+    public function consumeWhiteSpaceExtractsCommentWithSurroundingWhitespace(string $whitespace): void
+    {
+        $commentText = 'A walk-on part in the war for a lead role in a cage?';
```
Comment (oliverklee) [inline tests/Unit/Parsing/ParserStateTest.php:255]: Ah, that's the reference. Got it. :-)

**Comment by JakeQZ** [comment]
> Maybe we can add some human-readable description explaining how the method works.

I've added some description to the DocBlock for the method.  I've also added a test to confirm non-whitespace is not consumed, and tightened up a type annotation.

**Comment by JakeQZ** [comment]
> > I hope [David Gilmour](https://en.wikipedia.org/wiki/David_Gilmour) will appreciate this.
> 
> I don't understand the connection to this PR. 😢 Can you help me out?

While writing the tests, I had Pink Floyd's [Wish You Were Hear](https://www.youtube.com/watch?v=K6qj09OHvjw) as an earworm, so when constructing some test comments, I put in some of the lyrics from it.  (It's akin to the [Hello Dolly example plugin](https://en-gb.wordpress.org/plugins/hello-dolly/) that comes preinstalled with WordPress.)

File: src/Parsing/ParserState.php (hunk 1 above)
Comment (JakeQZ) [reply src/Parsing/ParserState.php:202]: I thought that higher levels just gave more warnings, rather than more deeply analyzing the code, though I may be wrong.  It's strange that it thinks that a method call which isn't a `get` won't modify the object.
